src/strategy/dungeons/wotlk/forgeofsouls/FosMultipliers.cpp

```diff
@@ -0,0 +1,55 @@
+#include "FosMultipliers.h"
+#include "FosActions.h"
+#include "GenericSpellActions.h"
+#include "ChooseTargetActions.h"
+#include "MovementActions.h"
+#include "FosTriggers.h"
+#include "FosActions.h"
+
+
+float BronjahmMultiplier::GetValue(Action* action) {
+    Unit* boss = nullptr;
+    boss = AI_VALUE2(Unit *, "find target", "bronjahm");
+    if (!boss)
+        return 1.0f;
+    if (botAI->IsTank(bot))
+        if (dynamic_cast<TankAssistAction*>(action))
+            return 0.0f;
+
+    if (boss && boss->HasUnitState(UNIT_STATE_CASTING) && boss->FindCurrentSpellBySpellId(SPELL_CORRUPT_SOUL))
+    {
+        if (dynamic_cast<MovementAction*>(action) && !dynamic_cast<MoveFromBronjahmAction*>(action))
+        {
+            return 0.0f;
+        }
+    }
+    return 1.0f;
+}
+
+float AttackFragmentMultiplier::GetValue(Action* action)
+{
+    if (botAI->IsHeal(bot) && dynamic_cast<AttackCorruptedSoulFragmentAction*>(action))
+        return 0.0f;
+
+    Unit* fragment = nullptr;
+
+    GuidVector targets = AI_VALUE(GuidVector, "possible targets no los");
+
+    for (auto i = targets.begin(); i != targets.end(); ++i)
+    {
+        Unit* unit = botAI->GetUnit(*i);
+        if (unit && unit->GetEntry() == NPC_CORRUPTED_SOUL_FRAGMENT)
+        {
+            fragment = unit;
+            break;
+        }
+    }
+
+    if (fragment && dynamic_cast<BronjahmDpsPositionAction*>(action))
```

Comment:
What are the purposes of these two checks - it seems like you are suppressing these two behaviours based on fragments being alive?